src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java

```diff
@@ -221,4 +235,79 @@ public MetadataRestriction setMetadataRestriction(Project project, MetadataTempl
 		return metadataRestrictionRepository.save(metadataRestrictionForFieldAndProject);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@PreAuthorize("hasPermission(#project, 'isProjectOwner')")
+	@Override
+	public List<MetadataTemplateField> getPermittedFieldsForRole(Project project, ProjectRole role) {
+		//get all fields for the project
+		List<MetadataTemplateField> metadataFieldsForProject = fieldRepository.getMetadataFieldsForProject(project);
+
+		//get all restrictions for the project
+		List<MetadataRestriction> restrictionForProject = metadataRestrictionRepository.getRestrictionForProject(
+				project);
+
+		//collect the fields into a map
+		Map<MetadataTemplateField, MetadataRestriction> restrictionMap = restrictionForProject.stream()
+				.collect(Collectors.toMap(MetadataRestriction::getField, field -> field));
+
+		//for each field to check
+		List<MetadataTemplateField> filteredFields = metadataFieldsForProject.stream()
+				.filter(field -> {
+					//if the restriction map contains the field
+					if (restrictionMap.containsKey(field)) {
+						MetadataRestriction metadataRestriction = restrictionMap.get(field);
+						ProjectRole level = metadataRestriction.getLevel();
+
+						//compare the restriction level to the given role.  If it's greater or equal, we're good
+						if (role.compareTo(level) >= 0) {
```

Comment:
Just a question on this part. Does it just lexicographically compare the strings in the enum. Example say the level is "PROJECT_OWNER" and the role is "PROJECT_USER", does it just compare these strings to each other? If that's the case it works for now but we will need to update this when more project roles are added

Reply:
Java's compareTo for an enum will order the objects in the order they're declared in the enum <https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Enum.html#compareTo(E)>.  Currently in ProjectRole it's specified as:
```java
public enum ProjectRole {

    PROJECT_USER("PROJECT_USER"),
    PROJECT_OWNER("PROJECT_OWNER");
```
I'm pretty sure you can't override compareTo in an enum in Java so regardless of the string names it'll stay in order as long as we order them correctly in the enum code.

Reply:
I think it might be more clear if we added a numeric value to the enum.

Reply:
@tom114 Thanks for the clarification. I didn't know exactly what was being compared there. I agree with @joshsadam that adding a numeric value to the enum would add some clarity

Reply:
Sure. Added a `level` property in ProjectRole so we can better compare those roles.  afc87f7

Reply:
That's perfect. Thanks @tom114